src/function/scalar/sequence/nextval.cpp

```diff
@@ -121,7 +121,9 @@ static unique_ptr<FunctionData> NextValBind(ClientContext &context, ScalarFuncti
 		// evaluate the constant and perform the catalog lookup already
 		auto seqname = ExpressionExecutor::EvaluateScalar(*arguments[0]);
 		if (!seqname.IsNull()) {
-			D_ASSERT(seqname.type().id() == LogicalTypeId::VARCHAR);
+			if (seqname.type().id() != LogicalTypeId::VARCHAR) {
+				seqname = seqname.CastAs(LogicalTypeId::VARCHAR, true);
+			}
 			auto qname = QualifiedName::Parse(StringValue::Get(seqname));
```

Comment:
It would be easier to call `seqname.ToString()` here rather than a cast followed by `StringValue::Get`

Reply:
Sure, done